Sources/NIO/NonBlockingFileIO.swift

```diff
@@ -214,6 +214,54 @@ public struct NonBlockingFileIO {
         }
     }
 
+    /// Write `byteCount` bytes from `buffer` to `fileHandle in `NonBlockingFileIO`'s private thread pool which is separate from any `EventLoop` thread.
+    ///
+    /// The returned `ByteBuffer` will not have less than `byteCount` bytes unless we hit end-of-file in which
+    /// case the `ByteBuffer` will contain the bytes available to read.
+    ///
+    /// - parameters:
+    ///   - fileHandle: The `FileHandle` to write to.
+    ///   - buffer: The `ByteBuffer` to write.
+    ///   - byteCount: The number of bytes to write from `buffer` to `fileHandle`.
+    ///   - eventLoop: The `EventLoop` to create the returned `EventLoopFuture` from.
+    /// - returns: An `EventLoopFuture` which is fulfilled if the write was successful or fails on error.
+    public func write(fileHandle: FileHandle,
+                      buffer: ByteBuffer,
+                      byteCount: Int,
+                      eventLoop: EventLoop) -> EventLoopFuture<()> {
+        guard byteCount > 0 else {
+            return eventLoop.newSucceededFuture(result: ())
+        }
+
+        var buf = buffer
+        return self.threadPool.runIfActive(eventLoop: eventLoop) {
+            var bytesWritten = 0
```

Comment:
We need to be a bit defensive here, as the user may as us to write 100 bytes from a byte buffer that only has 50 readable bytes, in which case we'll enter a nasty loop forever. We should create a local `byteCount = min(buffer.readableBytes, byteCount)`.

Reply:
In fact once we do that we don't need a `bytesWritten` local variable, because we can just subtract from `byteCount`.